geocoder/src/main/scala/grasshopper/api/Service.scala

```diff
@@ -0,0 +1,50 @@
+package grasshopper.api
+
+import java.net.InetAddress
+import java.time.Instant
+import scala.concurrent.ExecutionContextExecutor
+import akka.actor.ActorSystem
+import akka.stream.ActorFlowMaterializer
+import org.slf4j.LoggerFactory
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.Logger
+import akka.event.LoggingAdapter
+import grasshopper.protocol.GrasshopperJsonProtocol
+import grasshopper.model.Status
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.model.StatusCodes._
+import akka.http.scaladsl.coding.{ Deflate, Gzip, NoCoding }
+import spray.json._
+import io.geojson.FeatureJsonProtocol._
+
+trait Service extends GrasshopperJsonProtocol {
+  implicit val system: ActorSystem
+
+  implicit def executor: ExecutionContextExecutor
+  implicit val materializer: ActorFlowMaterializer
+
+  def config: Config
+
+  val logger: LoggingAdapter
+
+  lazy val log = Logger(LoggerFactory.getLogger("grashopper-geocoder"))
+
+  val routes = {
+    path("status") {
```

Comment:
This `/status` has the potential to be a bit more interesting than the fine-grained APIs.  Potentially this could call all other APIs' `/status` endpoints, returning `OK` if all its children are `OK`, otherwise returning:

* `WARN` if some child services are down/slow/etc., but parent API is "functional enough" to return results.
*  `DOWN` if enough child services are down to render the parent API unusable.

No real issue here, just thinking out loud for the future.